tests/pytests/test_aggregate_count.py

```diff
@@ -0,0 +1,416 @@
+        # (['FT.AGGREGATE', 'idx', '*', 'WITHOUTCOUNT', 'SORTBY', '2', '@price', 'ASC'], 10), # crash
+
+        # # WITHOUTCOUNT + SORTBY + LIMIT
+        (['FT.AGGREGATE', 'idx', '*', 'WITHOUTCOUNT', 'SORTBY', 1, '@title', 'LIMIT', 0, 50], 50),
+        (['FT.AGGREGATE', 'idx', '*', 'WITHOUTCOUNT', 'SORTBY', 1, '@price', 'LIMIT', 0, 50], 50),
+        (['FT.AGGREGATE', 'idx', '*', 'WITHOUTCOUNT', 'SORTBY', 1, '@title', 'LIMIT', 0, int(docs/2)], int(docs/2)),
+        (['FT.AGGREGATE', 'idx', '*', 'WITHOUTCOUNT', 'SORTBY', 1, '@title', 'LIMIT', 0, docs*4], docs),
+        (['FT.AGGREGATE', 'idx', '*', 'WITHOUTCOUNT', 'SORTBY', 1, '@title', 'LIMIT', 10, 50], 50),
+        (['FT.AGGREGATE', 'idx', '*', 'WITHOUTCOUNT', 'SORTBY', 1, '@title', 'LIMIT', 100, docs], docs - 100),
+        (['FT.AGGREGATE', 'idx', '@price:[1, 100]', 'WITHOUTCOUNT', 'SORTBY', 1, '@title', 'LIMIT', 0, docs], 100),
+    ]
+
+    for query, expected_results in queries_and_results:
+        cmd=' '.join(str(x) for x in query)
+
+        # create a query without WITHOUTCOUNT to test the default behavior
+        query_default = query.copy()
+        query_default.remove('WITHOUTCOUNT')
+        cmd_default=' '.join(str(x) for x in query_default)
+        res_default = env.cmd(*query_default)
+
+        for dialect in [1, 2, 3, 4]:
+            env.expect('CONFIG', 'SET', 'search-default-dialect', dialect).ok()
+            res = env.cmd(*query)
+
+            results = _get_results(res)
+            results_default = _get_results(res_default)
+
+            # Verify only the length of results, don't verify total_results
+            env.assertEqual(
+                len(results), expected_results,
+                message=f'{cmd}: len(results) != expected - dialect: {dialect}')
+
+            # Compare with the query without WITHOUTCOUNT
+            env.assertEqual(
+                len(results_default), expected_results,
+                message=f'{cmd_default}: len(results) != results_default - dialect: {dialect}')
+
+def test_withoutcount_resp3():
+    _test_withoutcount(3)
+
+def test_withoutcount_resp2():
+    _test_withoutcount(2)
+
+
+def _test_profile(protocol):
+    env = Env(protocol=protocol)
+    docs = 5
+    _setup_index_and_data(env, docs)
+
+    queries_and_profiles = [
+        # query,
+        # RESP2 Standalone,
+        # RESP3 standalone,
+        # RESP2 [shard[0], coordinator]
+        # RESP3 [shard[0], coordinator]
+
+        # WITHCOUNT
+        # No sorter, no limit, returns all results
+        (['FT.AGGREGATE', 'idx', '*', 'WITHCOUNT'],
+         ['Index', 'Sync Depleter'],
+         ['Index'],
+         [['Index', 'Sync Depleter'], ['Network', 'Sync Depleter']],
+         [['Index'], ['Network']]),
+
+        # WITHCOUNT + LIMIT 0 0
+        (['FT.AGGREGATE', 'idx', '*', 'WITHCOUNT', 'LIMIT', 0, 0],
+         ['Index', 'Counter'],
+         ['Index', 'Counter'],
+         [['Index', 'Sync Depleter'], ['Network', 'Counter']],
```

Comment:
If COUNT is also depleting, no need to add a depleter here?

Reply:
Count is depleting, so no depleter needed.